### PR TITLE
Faster way to update reading progress when using Log

### PR DESCRIPTION
**Pull request by skipadu**
When the book is in "Reading" state and user goes to "Log", the current page number is automatically focused and selected. Therefore the user can enter the new page right away instead of selecting and clearing the old one first.
Little faster way to update the reading progress.

What do you think about this improvement?

Screenshot
focusandautoselect_currentpage

### CODE REVIEW

**Comment by AndrewBennet** [comment]
You're on a roll! I will take a look at this soon 👍

File: ReadingList/ViewControllers/Edit/EditBookReadState.swift [hunk 1]
```diff
@@ -99,6 +99,15 @@ class EditBookReadState: FormViewController {
         monitorThemeSetting()
     }
 
+  override func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
+    if self.book.readState == .reading {
+      let currentPageRow = self.form.rowBy(tag: "currentPage") as! IntRow
```
Comment (AndrewBennet) [inline ReadingList/ViewControllers/Edit/EditBookReadState.swift]:
```suggestion
      let currentPageRow = self.form.rowBy(tag: currentPageKey) as! IntRow
```

File: ReadingList/ViewControllers/Edit/EditBookReadState.swift [hunk 2]
```diff
@@ -99,6 +99,15 @@ class EditBookReadState: FormViewController {
...
+      currentPageRow.cell.textField.selectAll(nil)
```
Comment (AndrewBennet) [inline ReadingList/ViewControllers/Edit/EditBookReadState.swift]:
```suggestion
```
I'm inclined to not pre-highlight the value of the field. I can imagine that causing some confusion and/or accidental deletion of the current value. Pre-selecting the field sounds good though.

File: ReadingList/ViewControllers/Edit/EditBookReadState.swift [hunk 3]
```diff
@@ -99,6 +99,15 @@ class EditBookReadState: FormViewController {
         monitorThemeSetting()
     }
 
+  override func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
+    if self.book.readState == .reading {
```
Comment (AndrewBennet) [inline ReadingList/ViewControllers/Edit/EditBookReadState.swift]:
```suggestion
    // If we are editing a book (not adding one), pre-select the current page field
    if self.book.readState == .reading && self.book.changedValues().isEmpty {
```
I think we shouldn't pre-select the field arriving on this page when adding a new book. To do that, I added this extra check. (If this page has been brought up from an "Edit" tap, the book will have no changedValues; if this is a new book in the process of being created, there will be numerous changedValues.)

File: ReadingList/ViewControllers/Edit/EditBookReadState.swift (hunk 2 above)
Comment (skipadu) [reply ReadingList/ViewControllers/Edit/EditBookReadState.swift]: Okay. That is probably the best way to make it reliable.

File: ReadingList/ViewControllers/Edit/EditBookReadState.swift (hunk 3 above)
Comment (skipadu) [reply ReadingList/ViewControllers/Edit/EditBookReadState.swift]: Good find. Didn't thought that when I implemented this.

**Comment by skipadu** [comment]
Okay. Haven't used the GitHub's pull requests or like any team feature beforee, so thanks for pointing that out. Will do the commit now.

I'm honored if you can show my name in the "credits" of next version :)

**Comment by AndrewBennet** [comment]
Ah there's actually an issue in one of my suggested changes - that `currentPageKey` variable isn't in scope in the function `viewDidAppear`. Not to worry - I'll merge this PR and fix it on master.

Are you on the TestFlight beta testing program? If not, you could sign up [here](https://testflight.apple.com/join/kBS5mVao) if you want; I will deploy a new version with this change soon.

**Comment by skipadu** [comment]
Okay 👍 
Yes, I'm already on the TestFlight beta testing :)

**Pull request merged**